### PR TITLE
DWPF-1291 Update Django to 4.2

### PR DESCRIPTION
**Pull request by alsyx**

### CODE REVIEW

**Review by CamLamb — APPROVED** [review]
LGTM!

**Comment by CamLamb** [comment]
Make sure to take a look at the release notes:
https://docs.djangoproject.com/en/5.0/releases/#id1
From 4.1.13 to 4.2.11

Note: Don't read through all of it, just any breaking changes/deprecations

**Comment by nicopicchio** [comment]
TODO -> Add new version of `.env.example` file

**Review by CamLamb — APPROVED** [review]
Make sure to run `make all-requirements` to update the requirements.txt file that the deployed instances use 😄

File: pyproject.toml [hunk 1]
```diff
@@ -11,9 +11,9 @@ python = "^3.12"
 Celery = "==5.2.7"
 crispy-forms-gds = "^0.2.3"
 dataclasses-json = "==0.5.7"
-django = "^4.1.10"
+django = "^4.2"
 djangorestframework = "^3.12.4"
-django-hawk-drf = "1.1.0"
+django-hawk-drf = "1.1.2"
```
Comment (CamLamb) [inline pyproject.toml:16]: Can we update this to 
```suggestion
django-hawk-drf = "^1.1.2"
```

File: pyproject.toml [hunk 2]
```diff
@@ -41,9 +41,9 @@ notifications-python-client = "^6.3.0"
 help-desk-client = "==0.2.0"
 govuk-bank-holidays = "^0.11"
 detect-secrets = "^1.4.0"
-dit-activity-stream = "^0.1.0"
 django-workflow-engine = "0.2.0"
 govuk-frontend-django = "0.3.7"
+dit-activity-stream = "0.2.3"
```
Comment (CamLamb) [inline pyproject.toml:46]: Can we update this to
```suggestion
dit-activity-stream = "^0.2.3"
```

File: pyproject.toml [hunk 3]
```diff
@@ -11,9 +11,9 @@ python = "^3.12"
 Celery = "==5.2.7"
 crispy-forms-gds = "^0.2.3"
 dataclasses-json = "==0.5.7"
-django = "^4.1.10"
+django = "^4.2"
```
Comment (CamLamb) [inline pyproject.toml:14]: Can we specify 4.2.11 here as this is the minimum security version we are willing to run
```suggestion
django = "^4.2.11"
```

**Review by CamLamb — APPROVED** [review]

**Pull request merged**